### PR TITLE
Initial usage of the `typescript-sdk`

### PR DESCRIPTION
**Pull request by andreaTP**
Based on #4338 

I updated everything on the Kiota side and converted one simple endpoint to use the TS SDK in the UI.

- should we move everything to Kiota?
- should we do it in a single PR? (there are 45 endpoints, it's doable but not nice to review)
- which `model`s should we use as a source of truth?
- what do you think about the approach for authentication?

### CODE REVIEW

**Comment by EricWittmann** [comment]
>     * should we move everything to Kiota?

Yes.

>     * should we do it in a single PR? (there are 45 endpoints, it's doable but not nice to review)

Let's get this PR merged first, then I'll convert everything else in a single PR.

>     * which `model`s should we use as a source of truth?

I'll delete all the existing models in favor of the Kiota generated ones.

>     * what do you think about the approach for authentication?

LGTM, although it's a bit hard to review `rest.utils.ts` code check warnings/errors.  LOL what editor are you using that is still using TAB characters?!

**Comment by andreaTP** [comment]
> LOL what editor are you using that is still using TAB characters?!

I told you I'm bad at formatting code :stuck_out_tongue: it's probably copy pasted.
I get this PR in a final shape and we can convert the remaining endpoints.
Warning: we "might" need some tweaks for files upload/download but let see ...

File: ui/package.json [hunk 1]
```diff
@@ -20,5 +20,8 @@
     "devDependencies": {
         "rimraf": "5.0.7",
         "shelljs": "0.8.5"
+    },
+    "dependencies": {
+        "@apicurio/apicurio-registry-client": "file:../typescript-sdk"
```
Comment (andreaTP) [inline ui/package.json:25]: this uses `npm link` ... not sure if we can improve it anyhow ...

File: ui/ui-app/src/utils/rest.utils.ts [hunk 2]
```diff
@@ -3,9 +3,63 @@ import { ContentTypes } from "@models/contentTypes.model.ts";
...
+export function createRegistryClient(config: ConfigService, auth: AuthService): ApicurioRegistryClient {
+    const authProvider = createAuthProvider(auth);
+    const adapter = new FetchRequestAdapter(authProvider);
+    adapter.baseUrl = config.artifactsUrl();
+    return createApicurioRegistryClient(adapter);
+}
+// END - Kiota Machinery
 
 
+const AXIOS = axios.create();
```
Comment (andreaTP) [inline ui/ui-app/src/utils/rest.utils.ts:61]: Ideally, at the end of the process, the rest of this file should go away.

File: ui/ui-app/src/utils/rest.utils.ts [hunk 3]
```diff
@@ -3,9 +3,63 @@ import { ContentTypes } from "@models/contentTypes.model.ts";
...
+            if (token) {
+                request.headers.add(TokenAuthenticationProvider.authorizationHeaderKey, `${this.key} ${token}`);
+            }
+        }
+    };
+}
+
+export function createAuthProvider(auth: AuthService): AuthenticationProvider {
+    if (auth.isOidcAuthEnabled()) {
+        return new TokenAuthenticationProvider("Bearer", async () => auth.getToken().then(v => v!));
```
Comment (andreaTP) [inline ui/ui-app/src/utils/rest.utils.ts:42]: Just to be sure, are we testing those configurations of the UI in CI?

**Comment by andreaTP** [comment]
I hope I have addressed all the comments, a couple of additional questions :slightly_smiling_face: :

- are we convinced about committing the generated code?
- is it ok to rebuild the `ts-sdk `in CI all the time?

File: ui/package.json (hunk 1 above)
Comment (EricWittmann) [reply ui/package.json:25]: ok - i'll see what I can do on this

File: ui/ui-app/src/utils/rest.utils.ts (hunk 3 above)
Comment (EricWittmann) [reply ui/ui-app/src/utils/rest.utils.ts:42]: What configurations?  We're only doing basic testing of the UI in CI - in two places.  We have UI tests in the registry repo, and we have some in the 3scale dev environment as well.

More UI testing is for sure needed - currents tests are mostly smoke tests.

Comment (andreaTP) [reply ui/ui-app/src/utils/rest.utils.ts:42]: Do we test the UI in an environment where OIDC is enables and Keycloak configured?

Comment (EricWittmann) [reply ui/ui-app/src/utils/rest.utils.ts:42]: In the 3scale dev environment, yes.  But not in registry CI.

**Comment by EricWittmann** [comment]
> I hope I have addressed all the comments, a couple of additional questions 🙂 :
> 
>     * are we convinced about committing the generated code?

I'm just getting to this now, but for sure we do **not** want to commit the generated code.  And this PR doesn't seem to do that...which is good.

> 
>     * is it ok to rebuild the `ts-sdk `in CI all the time?

**Comment by andreaTP** [comment]
> And this PR doesn't seem to do that...which is good.

I was mistaken due to the LOC changed :sweat_smile: .

> We could optimize I guess - only generate if the openapi.json file changes. But I don't think we're doing that for other SDKs, are we?

No, I think it's fine with this proposed setup.

**Comment by EricWittmann** [comment]
Closing in favor of https://github.com/Apicurio/apicurio-registry/pull/4806

**Pull request closed without merging**